web3_proxy/src/frontend/authorization.rs

```diff
@@ -1187,8 +1234,43 @@ impl Web3ProxyApp {
                                 None
                             };
 
-                        let rpc_key_id =
-                            Some(rpc_key_model.id.try_into().expect("db ids are never 0"));
+                        // Get the user_tier
+                        let user_model = user::Entity::find_by_id(rpc_key_model.user_id)
+                            .one(db_replica.conn())
+                            .await?
+                            .context("user model was not found, but every rpc_key should have a user")?;
+
+                        let balance = balance::Entity::find()
+                            .filter(balance::Column::UserId.eq(user_model.id))
+                            .one(db_replica.conn())
+                            .await?
+                            .context("related balance was not found but every user should have a balance")?
+                            .available_balance;
+
+                        let mut user_tier_model =
+                            user_tier::Entity::find_by_id(user_model.user_tier_id)
+                                .one(db_replica.conn())
+                                .await?
+                                .context("related user tier not found, but every user should have a tier")?;
+
+                        // TODO: Do the logic here, as to how to treat the user, based on balance and initial check
+                        // Clear the cache (not the login!) in the stats if a tier-change happens (clear, but don't modify roles)
+                        if user_tier_model.title == "Premium" && balance < Decimal::from(10) {
```

Comment:
Lets do `if user_tier_model.downgrade_tier_id.is_some() && balance < Decimal::from(0.01)` (or instead of `balance <= 0.01`, do `deposits <= spend`)

Requiring a $10 minimum deposit but then de-activating the account at $10 is probably not what we want. If a user runs a tiny bit negative, I don't think its much of a problem.

Reply:
that makes sense, added this